.github/workflows/cloud_build_failure_reporter.yml

```diff
@@ -0,0 +1,162 @@
+
+                  // check if any reporter opened any issues previously
+                  const prevIssues = await github.paginate(github.rest.issues.listForRepo, {
+                    ...context.repo,
+                    state: 'open',
+                    creator: 'github-actions[bot]',
+                    labels: [periodicLabel]
+                  });
+                  // createOrCommentIssue creates a new issue or comments on an existing issue.
+                  const createOrCommentIssue = async function (title, txt) {
+                    if (prevIssues.length < 1) {
+                      console.log('no previous issues found, creating one');
+                      await github.rest.issues.create({
+                        ...context.repo,
+                        title: title,
+                        body: txt,
+                        labels: [periodicLabel]
+                      });
+                      return;
+                    }
+                    if (prevIssues.length > 1) {
+                      console.warn(
+                        `found ${prevIssues.length} issues but only adding comment to ${prevIssues[0].html_url}`
+                      );
+                    }
+                    console.log(
+                      `found previous issue ${prevIssues[0].html_url}, adding comment`
+                    );
+                    await github.rest.issues.createComment({
+                      ...context.repo,
+                      issue_number: prevIssues[0].number,
+                      body: txt
+                    });
+                  };
+
+                  // updateIssues comments on any existing issues. No-op if no issue exists.
+                  const updateIssues = async function (txt) {
+                    if (prevIssues.length < 1) {
+                      console.log('no previous issues found.');
+                      return;
+                    }
+                    for (const prevIssue of prevIssues) {
+                      console.log(`found previous issue ${prevIssue.html_url}, adding comment`);
+                      await github.rest.issues.createComment({
+                        ...context.repo,
+                        issue_number: prevIssue.number,
+                        body: txt
+                      });
+                    }
+                  };
+
+                  // Find status of check runs.
+                  // We will find check runs for each commit and then filter for the periodic.
+                  // Checks API only allows for ref and if we use main there could be edge cases where
+                  // the check run happened on a SHA that is different from head.
+                  const commits = await github.paginate(github.rest.repos.listCommits, {
+                    ...context.repo
+                  });
+
+                  const relevantChecks = [];
+                  for (const commit of commits) {
```

Comment:
We are still looping through all commits. We should break out of this loop once we found both checks

Reply:
Added break loop logic when all checks are found.